code/game/objects/structures/barsign.dm

```diff
@@ -52,68 +49,76 @@
-		return
-	if(broken)
-		to_chat(user, "<span class ='danger'>The controls seem unresponsive.</span>")
+/obj/structure/sign/barsign/attack_hand(mob/user)
+	if(panel_open)
+		if(broken)
+			to_chat(user, "<span class='danger'>The controls seem unresponsive.</span>")
+			return
+		else
+			pick_sign()
+			to_chat(user, "<span class='notice'>You set the barsign and close the maintenance panel.</span>")
+			panel_open = FALSE
+			return
+	else
+		to_chat(user, "<span class='info'>The maintenance panel is currently closed.</span>")
 		return
-	pick_sign()
-
-
-
 
-/obj/structure/sign/barsign/attackby(var/obj/item/I, var/mob/user)
-	if( istype(I, /obj/item/screwdriver))
-		if(!panel_open)
-			to_chat(user, "<span class='notice'>You open the maintenance panel.</span>")
-			set_sign(new /datum/barsign/hiddensigns/signoff)
-			panel_open = 1
-		else
-			to_chat(user, "<span class='notice'>You close the maintenance panel.</span>")
-			if(!broken && !emagged)
-				set_sign(pick(barsigns))
-			else if(emagged)
-				set_sign(new /datum/barsign/hiddensigns/syndibarsign)
-			else
-				set_sign(new /datum/barsign/hiddensigns/empbarsign)
-			panel_open = 0
 
+/obj/structure/sign/barsign/attackby(var/obj/item/I, mob/user)
 	if(istype(I, /obj/item/stack/cable_coil) && panel_open)
 		var/obj/item/stack/cable_coil/C = I
-		if(emagged) //Emagged, not broken by EMP
+		if(broken && emagged) //Emagged, not broken by EMP
 			to_chat(user, "<span class='warning'>Sign has been damaged beyond repair!</span>")
 			return
-		else if(!broken)
+		else if(broken && !emagged)
+			if(C.use(2))
+				to_chat(user, "<span class='notice'>You replace the burnt wiring.</span>")
+				broken = FALSE
+			else
+				to_chat(user, "<span class='warning'>You need at least two lengths of cable!</span>")
+		else if(!broken && !emagged)
 			to_chat(user, "<span class='warning'>This sign is functioning properly!</span>")
 			return
+	else
+		return
 
-		if(C.use(2))
-			to_chat(user, "<span class='notice'>You replace the burnt wiring.</span>")
-			broken = 0
+/obj/structure/sign/barsign/screwdriver_act(mob/user)
+	if(allowed(user))
+		if(!panel_open)
+			to_chat(user, "<span class='notice'>You open the maintenance panel.</span>")
+			set_sign(new /datum/barsign/signoff)
+			panel_open = TRUE
 		else
-			to_chat(user, "<span class='warning'>You need at least two lengths of cable!</span>")
+			to_chat(user, "<span class='notice'>You close the maintenance panel.</span>")
+			panel_open = FALSE
 	else
-		return ..()
-
+		to_chat(user, "<span class='info'>Access denied.</span>")
+		return
 
+/obj/structure/sign/barsign/proc/pick_sign()
+	var/list/signs = barsigns
+	var/new_sign
+	if(!broken)
+		if(!emagged)
+			new_sign = input("Available Signage: ", "Bar Sign", null) as null|anything in signs
+			set_sign(new_sign)
+		else
+			signs += hiddensigns
+			new_sign = input("Available Signage: ", "Bar Sign", null) as null|anything in signs
+			set_sign(new_sign)
```

Comment:
On line 106, I think we need: `var/list/signs = barsigns.Copy()`
This is to avoid creating an alias for `barsigns`, because on line 113, we add to the list.